lib/algolia_index.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'algolia_index/models/firm'
+require_relative 'algolia_index/models/adviser'
+require_relative 'algolia_index/models/office'
+
+require_relative 'algolia_index/serializers/firm_serializer'
+require_relative 'algolia_index/serializers/adviser_serializer'
+require_relative 'algolia_index/serializers/office_serializer'
+
+module AlgoliaIndex
+  INDICES = {
+    advisers: "firm-advisers#{'-test' if Rails.env.test?}",
+    offices: "firm-offices#{'-test' if Rails.env.test?}"
+  }.freeze
+
+  class << self
+    def handle_update!(klass:, id:, firm_id:)
+      record = "AlgoliaIndex::#{klass}".constantize.new(
+        klass: klass,
+        id: id,
+        firm_id: firm_id
+      )
+      if record.exists?
```

Comment:
This internal API is slightly confusing unless you read the implementation. I think it would help to make some of these variables/method names more explicit. For example:

`search_record = "AlgoliaIndex::#{klass}".constantize.new`

`if search_record.db_record_exists?`

Reply:
@malcolmbaig Good point, glad you got the difference in type of record and proposed an alternative naming.

Reply:
♻️ :)